Realm/RLMRealm.mm

```diff
@@ -229,29 +241,25 @@ static void waitForPartialSyncSubscriptions(Realm::Config const& config) {
     row.set(expires_at_col, Timestamp(0, 0));
     realm->commit_transaction();
 
-    NotificationToken token;
-    Results results(realm, table);
-    CFRunLoopRef runLoop = CFRunLoopGetCurrent();
-    CFRunLoopPerformBlock(runLoop, kCFRunLoopDefaultMode, [&]() mutable {
-        token = results.add_notification_callback([&](CollectionChangeSet const&, std::exception_ptr) mutable {
-            if (table->size() > 1) {
-                token = {};
-                CFRunLoopStop(runLoop);
-            }
-        });
+    struct State {
+        std::shared_ptr<Realm> realm;
+        Results results;
+        NotificationToken token;
+    };
+    auto state = std::make_shared<State>(State{realm, Results(realm, table), {}});
+    state->token = state->results.add_notification_callback([=](CollectionChangeSet const&, std::exception_ptr) {
+        if (table->size() > 1) {
+            state->token = {};
+            completion();
+        }
     });
-    CFRunLoopRun();
 #else
+    static_cast<void>(queue);
     static_cast<void>(config);
+    static_cast<void>(completion);
```

Comment:
[q] What is the point of enabling this method if sync is not enabled?

Reply:
The function which calls this function exists in non-sync builds, so the callsite would need to be ifdefed out too if no definition was provided.